symphony/lib/toolkit/class.general.php

```diff
@@ -26,6 +27,24 @@ public static function sanitize($source)
         return $source;
     }
 
+    /**
+     * Convert any special characters into their entity equivalents.
+     * Contrary to `sanitize()`, this version does double encode existing entities.
+     *
+     * @since Symphony 2.7.5
+     * @uses htmlspecialchars()
+     * @param string $source
+     *  a string to operate on.
+     * @return string
+     *  the fully encoded version of the string.
+     */
+    public static function sanitizeDouble($source)
+    {
+        $source = htmlspecialchars($value, ENT_COMPAT, 'UTF-8', true);
```

Comment:
`$value` is wrong, it must be `$source` — this is a serious bug.

Reply:
fixed in #2823